platform/ios/src/MGLMapView.mm

```diff
@@ -2447,16 +2491,23 @@ - (void)addAnnotations:(NS_ARRAY_OF(id <MGLAnnotation>) *)annotations
         
         for (size_t i = 0; i < annotationTags.size(); ++i)
         {
-            MGLAnnotationTag annotationTag = annotationTags[i];
-            MGLAnnotationImage *annotationImage = annotationImages[i];
-            annotationImage.styleIconIdentifier = @(points[i].icon.c_str());
-            id <MGLAnnotation> annotation = annotations[i];
+            id<MGLAnnotation> annotation = annotations[i];
+            NSValue *annotationValue = [NSValue valueWithNonretainedObject:annotation];
```

Comment:
The existing code is already more indirect than it should be: we should build the `MGLAnnotationContext`s upfront, in the loop up above, and store them in a `vector`. We can assume that `Map::addPointAnnotations()` adds the annotations in order, so this loop would only need to associate the new tags with their annotation contexts.